platform/linux-generic/arch/default/odp_atomic.h

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2021, Arm Limited
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+#ifndef ODP_DEFAULT_ATOMIC_H_
+#define ODP_DEFAULT_ATOMIC_H_
+
+#ifdef __SIZEOF_INT128__
+
+typedef unsigned __int128 _u128_t;
+
+static inline _u128_t atomic_load_u128(_u128_t *atomic)
+{
+	return __atomic_load_n(atomic, __ATOMIC_RELAXED);
+}
+
+static inline int atomic_cas_acq_rel_u128(_u128_t *atomic,
+					  _u128_t old_val,
+					  _u128_t new_val)
+{
+	return __atomic_compare_exchange_n(atomic, &old_val, new_val,
+					   0 /* strong */,
+					   __ATOMIC_ACQ_REL,
+					   __ATOMIC_RELAXED);
+}
+
+static inline int atomic_is_lockfree_u128(void)
+{
+	return __atomic_is_lock_free(16, NULL);
+}
```

Comment:
These generic variants can now be implemented using standard  ODP API 128-bit atomics.

Reply:
On a second thought, this is simpler as the types don't have to be changed. Looks like the PR only needs a rebase and it's ready to be merged.

Reply:
I'd suggest to rename these functions while moving into header files. Otherwise, we loose a bit the context why queue_lf.c uses these functions and not the new 128bit API. Functions could be renamed like this:
atomic_load_u128 -> lockfree_load_u128
atomic_cas_acq_rel_u128 -> lockfree_cas_acq_rel_u128
atomic_is_lockfree_u128 -> lockfree_check_u128